cockatrice/src/window_main.cpp

```diff
@@ -86,7 +86,12 @@ void MainWindow::processConnectionClosedEvent(const Event_ConnectionClosed &even
 
 void MainWindow::processServerShutdownEvent(const Event_ServerShutdown &event)
 {
-    QMessageBox::information(this, tr("Scheduled server shutdown"), tr("The server is going to be restarted in %n minute(s).\nAll running games will be lost.\nReason for shutdown: %1", "", event.minutes()).arg(QString::fromStdString(event.reason())));
+    if (serverShutdownMessageBox)
+        serverShutdownMessageBox->close();
+    serverShutdownMessageBox = new QMessageBox(this);
```

Comment:
You leak memory here; the old messagebox will never gets deleted (unless you set Qt::WA_DeleteOnClose on it).
What about dropping the pointer?